archaius2-core/src/main/java/com/netflix/archaius/readers/PropertiesConfigReader.java

```diff
@@ -32,38 +35,46 @@
     private String              includeKey      = "@next";
 
     @Override
-    public Config load(ClassLoader loader, String resourceName) throws ConfigException {
+    public Config load(ClassLoader loader, String resourceName, StrInterpolator strInterpolator, StrInterpolator.Lookup lookup) throws ConfigException {
         URL url = getResource(loader, resourceName + ".properties");
         if (url == null) {
             throw new ConfigException("Unable to resolve URL for resource " + resourceName);
         }
-        return load(loader, url);
+        return load(loader, url, strInterpolator, lookup);
     }
 
     @Override
-    public Config load(ClassLoader loader, URL url) throws ConfigException {
+    public Config load(ClassLoader loader, URL url, StrInterpolator strInterpolator, StrInterpolator.Lookup lookup) throws ConfigException {
```

Comment:
K, just submitted a fix to switch to depth first instead of width first loading. Also adding the check for the infinite chaining.